File: scheduler-backend/src/main/java/webapp/scheduler/service/TaskService.java
package webapp.scheduler.service;

import java.util.List;

import webapp.scheduler.model.dto.TaskDTO;

public interface TaskService {

	List<TaskDTO> findAll();
	
	TaskDTO findById(Long id);
	
	List<TaskDTO> findByTypeId(Long id);
	
	TaskDTO create(TaskDTO taskDTO);
	
	TaskDTO update(TaskDTO taskDTO);
	
	TaskDTO deleteById(Long id);
	
}

File: scheduler-frontend/src/services/task-service.js
import { TASK_API_URL } from "../constants/API-constants";
import apiHelper from "../utils/api-helper";

const API_URL = TASK_API_URL;

class TaskService {

  findAll() {
    return apiHelper.sendGetRequest(`${API_URL}/all`);
  }

  findById(id) {
    return apiHelper.sendGetRequest(`${API_URL}?id=${id}`);
  }

  create(task) {
    return apiHelper.sendPostRequest(`${API_URL}`, task);
  }

  update(task) {
    return apiHelper.sendPutRequest(`${API_URL}`, task);
  }

  deleteById(id) {
    return apiHelper.sendDeleteRequest(`${API_URL}?id=${id}`);
  }
}

export default new TaskService();